### PR TITLE
Refactor gateway gets pre events

### PR DESCRIPTION
**Pull request by jebbinSCL**

### CODE REVIEW

File: Gateway/CorrespondenceController.fs [hunk 1]
```diff
@@ -1,18 +1,15 @@
...
     member x.Get(senderId : string, receiverId : string) = 
-        Log.Information("Received get request for correspondence between sender: {senderId} and reciever : {recieverId}", senderId, receiverId) 
-        match getCorrespondence(senderId, receiverId) with
-          | Success correspondence -> x.Request.CreateResponse(correspondence)
-          | Failure error -> x.Request.CreateResponse(error.HttpStatusCode, error.Body)
-  
\ No newline at end of file
+        (fun () -> 
+            let correspondence = getCorrespondence senderId receiverId
+            correspondence |> Seq.map Correspondence.toModel)
```
Comment (pallenSL) [inline Gateway/CorrespondenceController.fs]: Why isn't this just
getCorrespondence senderId receiverId |> Seq.map Correspondence.toModel

I'm not suggesting it's wrong; it's just not the beautiful, flowing, near-perfect English that that website promised.

File: Gateway/CorrespondenceProxy.fs [hunk 2]
```diff
@@ -1,51 +1,10 @@
...
-    with
-        | :? AggregateException ->
-            Log.Information("Could not reach correspondence endpoint")
-            Failure { HttpStatusCode = HttpStatusCode.InternalServerError; Body = "Could not reach correspondence endpoint" }
-        | ex ->
-            Log.Information("Unhandled exception: {message}", ex.Message)
-            Failure { HttpStatusCode = HttpStatusCode.InternalServerError; Body = "An unhandled error occurred: " + ex.Message }
+let getCorrespondence (senderId : string) (receiverId : string) = 
+    let uri = Uri <| String.Format("{0}?profileIdOne={1}&profileIdTwo={2}", correspondenceUrl, senderId, receiverId)
+    get<CorrespondenceItem []>(uri)
```
Comment (pallenSL) [inline Gateway/CorrespondenceProxy.fs]: Please mister, can we have a newline?

File: Gateway/CorrespondenceController.fs (hunk 1 above)
Comment (jebbinSCL) [inline Gateway/CorrespondenceController.fs]: Which website? I think that was some mix between rhetoric and promise, rather than 100% promise.

**Comment by dmorrisSL** [comment]
:shipit:

**Comment by dmorrisSL** [comment]
Merging -- GitHub notifications are down at the moment. @jebbin has slack-confirmed that he's addressed @pallen's comment

**Pull request merged**